libhb/hb.c

```diff
@@ -696,7 +696,13 @@ hb_buffer_t * hb_read_preview(hb_handle_t * h, hb_title_t *title, int preview)
 
         for (hh = 0; hh < h; hh++)
         {
-            fread(data, w, 1, file);
+            if (fread(data, w, 1, file) < w)
+            {
+                hb_error( "hb_read_preview: Failed to read line %d from %s" , hh, filename );
+                hb_buffer_close(&buf);
+                break;
+            }
```

Comment:
You may have noticed this particular commit was recently reverted due to crashes.

You're freeing the buffer, and then closing the inner loop:

for (hh = 0; hh < h; hh++)

…yet the outer loop:

for (pp = 0; pp < 3; pp++)

…uses buf without checking for NULL. While this is a pre-exisiting error, breaking the outer loop (may with a goto label just before fclose, else fclosing within the error condition and returning NULL) is within the scope of your patch.

In addition, it seems fread will return less than w in real-world conditions. While this may be yet another bug, I notice you assume every case where fread doesn't return size * nitems is an error, and don't bother checking whether we had a read error vs. EOF.

If the purpose of your patch is to clear warnings for its own sake, then it shouldn't attempt to be smart and just store the result of fread somewhere without using it. But since you seem willing to go the extra mile, then maybe you should check for ferror(file) and error out only if it is true (not just here but elsewhere in the patch).

Reaching EOF early may indeed be an error and/or a bug elsewhere in the code, but probably not worth ditching the whole preview for (as it most likely happens near the end).

Regards,

Tim